### PR TITLE
Add redact verb to CLI

### PR DESCRIPTION
**Pull request by sfoslund**
Adding the new `redact` verb to the sbom tool CLI. For now this verb is excluded from the tool help since the actual redaction logic is not yet present. With these changes running redact simply prints the arguments provided for the command, but redaction logic will be added in a future PR. 

Once we turn this feature on it will show up in the help text as follows: 
```  
  Redact -options - Redact sensitive file information from given SBOM(s).

    Option            Description
    SbomPath (-sp)    The file path of the SBOM to redact.
    SbomDir (-sd)     The directory containing the sbom(s) to redact.
    OutputPath (-o)   Gets or sets the directory where the redacted SBOM file(s) will be generated.
    Verbosity (-V)    Display this amount of detail in the logging output.
                      Verbose
                      Debug
                      Information
                      Warning
                      Error
                      Fatal

  Version  - Displays the version of the tool being used. Can be used as '--version'
```

This PR also updates the style linter to be less strict about some comment style rules.

### CODE REVIEW

File: src/Microsoft.Sbom.DotNetTool/RedactService.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    private readonly IRecorder recorder;
+
+    private readonly IHostApplicationLifetime hostApplicationLifetime;
+
+    public RedactService(
+        IWorkflow<SbomRedactionWorkflow> redactionWorkflow,
+        IRecorder recorder,
+        IHostApplicationLifetime hostApplicationLifetime)
+    {
+        this.redactionWorkflow = redactionWorkflow;
```
Comment (DaveTryon) [inline src/Microsoft.Sbom.DotNetTool/RedactService.cs:29]: Should these be validated against null values?

File: src/Microsoft.Sbom.Tool/RedactService.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    private readonly IHostApplicationLifetime hostApplicationLifetime;
+
+    public RedactService(
+        IWorkflow<SbomRedactionWorkflow> redactionWorkflow,
+        IRecorder recorder,
+        IHostApplicationLifetime hostApplicationLifetime)
+    {
+        this.redactionWorkflow = redactionWorkflow;
+        this.recorder = recorder;
+        this.hostApplicationLifetime = hostApplicationLifetime;
```
Comment (DaveTryon) [inline src/Microsoft.Sbom.Tool/RedactService.cs:27]: Should these arguments be validated against null?

File: .editorconfig [hunk 3]
```diff
@@ -496,6 +496,15 @@ dotnet_diagnostic.CA1863.severity = none
...
 
+# SA1005: Single line comments should begin with single space
+dotnet_diagnostic.SA1005.severity = none
+
+# SA1515: Single-line comment should be preceded by blank line
+dotnet_diagnostic.SA1515.severity = none
+
+# SA1512: Single-line comments should not be followed by blank line
+dotnet_diagnostic.SA1512.severity = none
+
```
Comment (DaveTryon) [inline .editorconfig:507]: Are these related to the changes? I don't notice any changes that stand out as being obvious matches

**Review by DaveTryon — APPROVED** [review]
Added a couple of questions, but seems OK to me

File: .editorconfig (hunk 3 above)
Comment (sfoslund) [reply .editorconfig:507]: No, this change just disables some very strict linting rules surrounding comments that were impeding my and Alison's dev loops.

Comment (alisonlomaka) [reply .editorconfig:507]: IMO if we want strict linting rules like these, we should be automatically applying them at some point, instead of either interrupting flow or breaking compile and forcing engs to fix them one at a time. So painful.

**Review by alisonlomaka — APPROVED** [review]
LGTM!
